lib/livebook_web/live/session_live.ex

```diff
@@ -550,11 +561,15 @@ defmodule LivebookWeb.SessionLive do
 
   @impl true
   def handle_params(%{"cell_id" => cell_id}, _url, socket) do
+    assert_live_action_access!(socket)
+    assert_policy!(socket, :edit)
```

Comment:
What I had in mind was to cover `assert_policy!` inside `assert_live_action_access!`, something like this:

```elixir
defp assert_live_action_access!(%{assigns: %{live_action: live_action}} = socket) do
  cond do
    live_action in [...] -> assert_policy!(socket, :edit)
    live_action in [...] -> assert_policy!(socket, :read)
    true -> raise "unknown live_action: #{inspect(live_action)}"
  end
end
```

This way `handle_params` can stay decoupled from `live_action` and we don't miss any route :)